deltachat-server/rpc-client.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import asyncio
+import json
+
+
+class Deltachat:
+    def __init__(self, process):
+        self.process = process
+        self.event_queue = asyncio.Queue()
+        self.id = 0
+        self.reader_task = asyncio.create_task(self.reader_loop())
+
+        # Map from request ID to asyncio.Event.
+        # When event happens, it is replaced with a response.
+        self.request_events = {}
+
+    async def reader_loop(self):
+        while True:
+            line = await self.process.stdout.readline()
+            response = json.loads(line)
+            if "id" in response:
+                response_id = response["id"]
+                event = self.request_events[response_id]
+                self.request_events[response_id] = response
+
+                # Notify that response arrived.
+                event.set()
+            else:
+                if response["method"] == "event":
+                    # An event notification.
+                    await self.event_queue.put(response["params"])
+                continue
+
+    async def get_next_event(self):
+        return await self.event_queue.get()
+
+    def __getattr__(self, attr):
+        async def method(*args):
```

Comment:
can we set the `__name__` of this correctly?

Reply:
maybe this "dynamic" creation of arbitrary methods is good for a first go but later we probably want to have an explicit list of methods and input parameters along with docstrings for each method.  Otherwise it's hard to know which methods one can call.  It might make sense to generate the according methods automatically from jsonrpc definitions but i don't know much about it yet.